mpv.go

```diff
@@ -253,38 +303,40 @@ func (m *Mpv) TerminateDestroy() {
 
 // convertData converts the data according to the given format and returns an unsafe.Pointer
 // for use in SetOption and SetProperty.
-func convertData(format Format, data interface{}) unsafe.Pointer {
+// Note: If conversion lead to a memory allocation, you need to defer call the returned function.
+func convertData(format Format, data interface{}) (unsafe.Pointer, func()) {
 	switch format {
 	case FORMAT_NONE:
-		return nil
+		return nil, nil
 	case FORMAT_STRING, FORMAT_OSD_STRING:
-		return unsafe.Pointer(&[]byte(data.(string))[0])
+		str := C.CString(data.(string))
+		return unsafe.Pointer(str), func() { C.free(unsafe.Pointer(str)) }
 	case FORMAT_FLAG:
-		var val int
+		var val C.int
 		if data.(bool) {
 			val = 1
 		} else {
 			val = 0
 		}
-		return unsafe.Pointer(&val)
+		return unsafe.Pointer(&val), nil
 	case FORMAT_INT64:
 		i, ok := data.(int64)
 		if !ok {
 			i = int64(data.(int))
 		}
 		val := C.int64_t(i)
-		return unsafe.Pointer(&val)
+		return unsafe.Pointer(&val), nil
 	case FORMAT_DOUBLE:
 		val := C.double(data.(float64))
-		return unsafe.Pointer(&val)
+		return unsafe.Pointer(&val), nil
 	case FORMAT_NODE:
```

Comment:
All the node ones would need a free func as well, but I'm not using them so I'm OK with merging as-is, and filing an issue to keep track that we still have a known memory leak